bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/discovery/PilightDeviceDiscoveryService.java

```diff
@@ -178,9 +179,13 @@ public void setThingHandler(final ThingHandler handler) {
 
     @Override
     public void activate() {
-        super.activate(null);
         final @Nullable PilightBridgeHandler pilightBridgeHandler = this.pilightBridgeHandler;
         if (pilightBridgeHandler != null) {
+            if (pilightBridgeHandler.isBackgroundDiscoveryEnabled()) {
+                super.activate(null);
+            } else {
+                removeOlderResults(new Date().getTime(), pilightBridgeHandler.getThing().getUID());
+            }
```

Comment:
I would prefer that you call super.activate in all cases, passing the property "background" with either value true or false.

Reply:
Done